File: CanPlaceFlowers/CanPlaceFlowers.py
from typing import List


def canPlaceFlowers(flowerbed: List[int], n: int) -> bool:
        flowers = 0
        for i in range(len(flowerbed)):
            if flowerbed[i] == 0:
                left = (i == 0) or (flowerbed[i - 1] == 0)
                right = (i == len(flowerbed) - 1) or (flowerbed[i + 1] == 0)

                if left and right:
                    flowerbed[i] = 1
                    flowers += 1
        
        return flowers >= n

flowerbed = [1,0,0,0,1]
n = 1

print(canPlaceFlowers(flowerbed,n))

File: CanPlaceFlowers/CanPlaceFlowers.java
package CanPlaceFlowers;

import java.util.*;

public class CanPlaceFlowers {
    public static boolean soltion(int[] flowerbed, int n) {
    
        int flowers = 0;
        for(int i = 0; i < flowerbed.length ; i++){
            if (flowerbed[i] == 0) {
            boolean left = (i == 0) || (flowerbed[i - 1] == 0);
            boolean right = (i == flowerbed.length - 1) || (flowerbed[i + 1] == 0);

            if(left && right){
                flowerbed[i] = 1;
                flowers ++;
                if (flowers >= n) {
                        return true;
                    }
            }
            }
        }
        return flowers >= n;
    }

    public static void main(String[] args) {
        int[] flowerbed = {1, 0, 0, 0, 1};
        int n = 1;


        System.out.println(soltion(flowerbed,n));
    }
}
